src/main/java/com/iota/iri/service/API.java

```diff
@@ -830,7 +831,8 @@ private AbstractResponse getNodeInfoStatement(){
                 System.currentTimeMillis(), 
                 instance.tipsViewModel.size(),
                 instance.transactionRequester.numberOfTransactionsToRequest(),
-                features
+                features,
+                instance.lagCalculator.getMedianArrivalLag()
```

Comment:
we should be aware that this is not O(1), but the rest of this call is; `getNodeInfo` is often used as a heartbeat and should return very fast.

I don't have a proposal on how to solve this, maybe having `lagCalculator` run a thread that calculates this metric every once in a while and answers the get in O(1).

Reply:
>there is no heavy lifting being done.
we have the sorted heap part that may be an issue, but I agree that we can address it in the future if needed.